crates/speedy-transform/src/web_transform/parser.rs

```diff
@@ -30,10 +30,16 @@ pub fn transform(
   let fm = cm.new_source_file(FileName::Custom(source_filename.clone()), code.into());
   let compiler = Compiler::new(cm);
 
+  #[cfg(not(target_arch = "wasm32"))]
+  let tsx_parse = config.tsx.unwrap_or(true);
+  // wasm plugin parse option is in js side, not in rust side
+  #[cfg(target_arch = "wasm32")]
+  let tsx_parse = true;
+
   let lexer = Lexer::new(
     // We want to parse ecmascript
     Syntax::Typescript(TsConfig {
-      tsx: true,
+      tsx: tsx_parse,
```

Comment:
For this PR, this newly added option looks good to me. 
However, It comes to my mind that we should not treat any input files as `TypeScript` formatted. For JavaScript-like sources like `js` or `jsx`, it should complain this behavior as an error. But we should think if we might need to add this in a backwards-compatable way.

Reply:
Maybe we should use config like `code_type: Option<String>`.
It can be `js`, `jsx`, `ts`, `tsx`, which means what type users want to parse. But of course, default and illegal string will use `tsx` parse.

Reply:
It will be better to use `enum`, but regarding of the current architecture with node binding and wasm. It looks good to me.